datasimulator/file_handling.py

```diff
@@ -0,0 +1,20 @@
+from .utils import attempt  # same as datasimulator.utils
+from cdislogging import get_logger
+logger = get_logger("data-simulator", log_level="info")
+
+
+def write_node_to_file(outfile, node):
+    outfile.write(f"{node.name}\t{node.category}\n")
+
+
+def write_to_file_or_log_error(file_path, submission_order):
+    file_open_result = attempt(lambda: open(file_path, "w"))
```

Comment:
here too i think this is a bit overengineered, in the sense that we add an extra layer that makes it harder to follow what the code is doing.

this is effectively the same, has fewer lines of code and is more straighforward to understand:
```
try:
  with open(...) as f:
    write to file
except:
  failure log
  raise
```
or if you really want to separate the failure to open the file:
```
try:
  f = open file
except:
  failure log
  raise
try:
  write to f
finally:
  close f
```

Reply:
I think I agree, but I'm still thinking about this. Will follow up with either a commit or another comment.

Reply:
I decided to just scrap most of it and revert back to a simple open. We don't need to do anything else if after this, so we can just crash and communicate any unexpected I/O errors if they happen. No try-catch needed.